libs/utils/trace.py

```diff
@@ -874,6 +867,27 @@ def getClusterActiveSignal(self, cluster):
 
         return cluster_active
 
+    def addEventsDeltas(self, df, col_name='len'):
+        """
+        Compute the time between each event in a dataframe, and store it in a
+        new column. This only really makes sense for events tracking an
+        on/off state (e.g. overutilized, idle)
+        """
+        if df.empty:
+            return df
+
+        if col_name in df.columns:
+            raise RuntimeError("Column {} is already present in the dataframe".
+                               format(col_name))
+
+        df['start'] = df.index
+        df[col_name] = (df.start - df.start.shift()).fillna(0).shift(-1)
+        df.drop('start', axis=1, inplace=True)
+
+        # Fix the last event, which will have a NaN duration
+        # Set duration to trace_end - last_event
+        df.loc[df.index[-1], 'len'] = self.start_time + self.time_range - df.index[-1]
```

Comment:
:thinking: ... should not `s/'len'/col_name/` ?

Reply:
:sweat: